LoopKit/CriticalEventLog.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  CriticalEventLog.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 7/15/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public protocol EstimatedDurationProgressor {
```

Comment:
The idea of cancelable progress and rolled up tasks seems really similar to the built in Progress class: https://developer.apple.com/documentation/foundation/progress

Reply:
On a larger scale; if this is happening in the background, do we need progress reporting?

Reply:
We need progress for the full export in case the nightly exports have (for whatever reason) not completed. It is an edge case, but it definitely needs progress. (Also, with my current work looking into a different zip implementation, it looks like it may take longer than 3-5 seconds for the full export to complete so we at least need a progress bar.)

I did look into `Progress` and at the time did not think it appropriate. I will revisit.

Reply:
Updated to use `Progress`. Required a number of downstream changes, but should be good now. ✅